doc/glossary.rst

```diff
@@ -96,4 +96,35 @@ Glossary of Terms
       term storage rather than copying it into RAM, saving writable memory for
       dynamic data and not the static program code.
 
+   Zephyr downstream module
+      A Zephyr module that the :file:`zephyr` repository default configuration
+      and default manifest file are **not** aware of.
+
+   Zephyr manifest module
+      A Zephyr module which also contains one or more west manifest files.
+
+   Zephyr module
+      A YAML file named :file:`zephyr/module.yml` and all the content it
+      references, which define the extensions to be hooked into the Zephyr
+      infrastructure.
+
+   Zephyr Project module
+      A Zephyr module managed by the Zephyr Project.  These modules will
+      be hosted at the Zephyr project organization on Github.
+
+   Zephyr upstream module
+      A Zephyr module that the :file:`zephyr` repository default configuration
+      and default manifest file are aware of.
+
+   :makevar:`ZEPHYR_MODULES`
+      The Zephyr build system CMake variable through which a list paths to
+      all Zephyr modules may be provided.
+
+   :makevar:`ZEPHYR_EXTRA_MODULES`
+      The Zephyr build system CMake variable through which paths to
+      Zephyr modules may be provided.  This is provided to support module
+      development tangential to how :makevar:`ZEPHYR_MODULES` is being used.
```

Comment:
or it can be used in combination to `west`, where `west` knows about the manifest defined modules, but you have additional local modules or private repos (as example for testing or confidential code) that you want to include in a build.

Could be, as example, when working with the source for a closed source stack.

Reply:
Good to know. Please suggest replacement text if you feel that needs to be captured.

Reply:
will do some thinking.